modyn/selector/internal/selector_manager.py

```diff
@@ -185,24 +186,16 @@ def uses_weights(self, pipeline_id: int) -> bool:
 
         return self._selectors[pipeline_id].uses_weights()
 
-    def _instantiate_strategy(self, selection_strategy: dict, pipeline_id: int) -> AbstractSelectionStrategy:
-        strategy_name = selection_strategy["name"]
-        maximum_keys_in_memory = selection_strategy["maximum_keys_in_memory"]
-        config = dict(selection_strategy)
-        default_configs = {"limit": -1, "reset_after_trigger": False}
-
-        for setting, default_value in default_configs.items():
-            if setting not in config:
-                config[setting] = default_value
-                logger.warning(f"Setting {setting} to default {default_value} since it was not given.")
-
+    def _instantiate_strategy(self, raw_selection_strategy: str, pipeline_id: int) -> AbstractSelectionStrategy:
+        selection_strategy = TypeAdapter(SelectionStrategyModel).validate_json(raw_selection_strategy)
+        strategy_name = selection_strategy.name
         strategy_module = dynamic_module_import("modyn.selector.internal.selector_strategies")
         if not hasattr(strategy_module, strategy_name):
             raise NotImplementedError(f"Strategy {strategy_name} not available!")
 
         strategy_handler = getattr(strategy_module, strategy_name)
 
-        return strategy_handler(config, self._modyn_config, pipeline_id, maximum_keys_in_memory)
+        return strategy_handler(selection_strategy, self._modyn_config, pipeline_id)
 
```

Comment:
`maximum_keys_in_memory` is already a field in `config`, we don't need to create a separate argument for it.